File: core/src/com/alta_v2/rendering/tiledMapScreen/layout/LayoutFactory.java
package com.alta_v2.rendering.tiledMapScreen.layout;

import com.alta_v2.rendering.tiledMapScreen.layout.map.MapLayoutImpl;
import com.alta_v2.rendering.tiledMapScreen.layout.person.NpcLayout;
import com.alta_v2.rendering.tiledMapScreen.layout.person.PlayerLayout;
import com.badlogic.gdx.assets.AssetManager;

import java.util.Map;

public interface LayoutFactory {

    PlayerLayout createPlayerLayout(AssetManager assetManager, String texturePath);

    NpcLayout createNpcLayout(AssetManager assetManager, Map<Integer, String> npcTextures);

    MapLayoutImpl createMapLayout(AssetManager assetManager, String mapPath);
}

File: core/src/com/alta_v2/rendering/tiledMapScreen/layout/LayoutFactory.kt
package com.alta_v2.rendering.tiledMapScreen.layout

import com.alta_v2.rendering.tiledMapScreen.layout.map.MapLayoutImpl
import com.alta_v2.rendering.tiledMapScreen.layout.person.NpcLayout
import com.alta_v2.rendering.tiledMapScreen.layout.person.PlayerLayout
import com.badlogic.gdx.assets.AssetManager

interface LayoutFactory {

    fun createPlayerLayout(assetManager: AssetManager, texturePath: String): PlayerLayout

    fun createNpcLayout(assetManager: AssetManager, npcTextures: Map<Int, String>): NpcLayout

    fun createMapLayout(assetManager: AssetManager, mapPath: String): MapLayoutImpl
}